src/Common/SettingConstraintType.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+
+namespace DB
+{
+
+enum class SettingConstraintType
+{
+    // Default. Behave in the same way as WRITABLE, but is not inherited unless `settings_constraints_replace_previous` is set.
+    NONE,
```

Comment:
Why do we need this `NONE` type? And use `WRITABLE` as the default?

Reply:
Can be done, see previous comment